cupyx/jit/_compile.py

```diff
@@ -464,13 +464,18 @@ def _transpile_stmt(stmt, is_toplevel, env):
     if isinstance(stmt, ast.AugAssign):
         value = _transpile_expr(stmt.value, env)
         target = _transpile_expr(stmt.target, env)
-        assert isinstance(target, Data)
+        if not isinstance(target, Data):
+            raise TypeError(f'Cannot augassign to {target.code}')
         value = Data.init(value, env)
-        result = _eval_operand(stmt.op, (target, value), env)
+        tmp = Data('tmp__', target.ctype)
```

Comment:
Just curious: Are there any other variable names that are reserved by CuPy JIT?

Reply:
The users might use `tmp__` as a local variable in the python function. Fixed in [1c229fa](https://github.com/cupy/cupy/pull/6844/commits/1c229fa0e9e2f63b39376594d80ce8b71dafcb24).

Reply:
Looks nice!